_source/_posts/2022-05-04-angular-microfrontend-deploy.md

```diff
@@ -0,0 +1,286 @@
+
+Security + testing = winning!
+
+## Build the micro-frontend on a relevant change
+
+As mentioned in this post, the value of micro-frontends is the ability to deploy each application independently. In production systems, you may need to handle multiple build steps and build out an entire workflow of CI/CD, but for this tutorial, we can cheat a little and still get the benefits of separate deployments. We will add a configuration in Vercel only to kick off a build when there's changes in the relevant application code.
+
+From the Vercel dashboard, open the Vercel project for the `mfe-basket` application. Navigate to the **Settings** tab and select **Git**. In the **Ignored Build Step**, we can add a Git command to ignore all changes except those in the `projects/mfe-basket` directory.
+
+Add the following command:
+
+```shell
+git diff --quiet HEAD^ HEAD ./projects/mfe-basket
+```
+
+Next, open the Vercel project for the `mfe-profile` application to update the **Ignored Build Step** command. Add the following command:
+
+```shell
+git diff --quiet HEAD^ HEAD ./projects/mfe-profile
+```
+
+Lastly, open the Vercel project for the `shell` application to update the **Ignored Build Step** command. This command is a little different because we want to pick up any changes to the project excluding this changes to `mfe-basket` and `mfe-profile`. Add the following command:
+
+```shell
+git diff --quiet HEAD^ HEAD -- ':!projects/mfe-basket' ':!projects/mfe-profile'
+```
+
+Now, if you make a change in a micro-frontend application, only that micro-frontend application will build and deploy. All three projects will notice the change and start building but will immediately cancel the build if the build step should be ignored for the project.
+
+## Beyond this post
+
+I hope you enjoyed creating and deploying this micro-frontend e-commerce site with beautiful pictures of tasty cupcakes. In this post, we used dynamic module loading to load micro-frontends lazily within the host application, deployed the project to Vercel, added in multi-environment route configuration, and leveraged a benefit of micro-frontend by enabling deployment upon relevant changes. You can check out the completed code for this project in the [`deploy` branch of the code repo](https://github.com/oktadev/okta-angular-microfrontend-example/tree/deploy).
```

Comment:
```suggestion
I hope you've enjoyed creating and deploying this micro-frontend e-commerce site with beautiful pictures of tasty cupcakes. In this post, we used dynamic module loading to load micro frontends lazily within the host application, deployed the project to Vercel, added in multi-environment route configuration, and leveraged a micro-frontend benefit by enabling deployment upon relevant changes. You can check out the completed code for this project in the [`deploy` branch of the code repo](https://github.com/oktadev/okta-angular-microfrontend-example/tree/deploy).
```
changed tense (you've enjoyed) to indicate an ongoing action rather than a single finite action